opus/application/apps/cart/views.py

```diff
@@ -72,27 +74,53 @@ def api_view_cart(request):
 
     This is a PRIVATE API.
 
-    Format: __cart/view.html
+    Format: __cart/view.json
+    Arguments: reqno=<reqno>
+               Normal search arguments
     """
+
     api_code = enter_api_call('api_view_cart', request)
 
     if not request or request.GET is None:
         ret = Http404(settings.HTTP404_NO_REQUEST)
         exit_api_call(api_code, ret)
         raise ret
+    session_id = get_session_id(request)
+
+    reqno = get_reqno(request)
+    if reqno is None:
+        log.error('api_view_cart: Missing or badly formatted reqno')
+        ret = Http404(settings.HTTP404_MISSING_REQNO)
+        exit_api_call(api_code, ret)
+        raise ret
 
     session_id = get_session_id(request)
+    get_not_selected_product_types_str = request.GET.get('unselected_types', '')
+    not_selected_product_types = get_not_selected_product_types_str.split(',')
 
-    product_types = ['all']
+    product_types_str = request.GET.get('types', 'all')
+    product_types = product_types_str.split(',')
 
     info = _get_download_info(product_types, session_id)
     count, recycled_count = get_cart_count(session_id, recycled=True)
 
+    for product in info['product_cat_list']:
```

Comment:
You can just say `for name, details in info`...

Reply:
got it.